test/router.spec.ts

```diff
@@ -0,0 +1,707 @@
+    }) as any)
+    await router.route(
+      {
+        body: qs.stringify({ name: 'tim' }),
+        headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+      } as any,
+      {},
+      '/route',
+      'POST'
+    )
+  })
+
+  test('route parses url-encoded body with multi-typed content header', async () => {
+    expect.assertions(1)
+    const router = Router()
+    router.post('/route', (({ body }) => {
+      expect(body.name).toBe('tim')
+    }) as any)
+    await router.route(
+      {
+        body: qs.stringify({ name: 'tim' }),
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+        },
+      } as any,
+      {},
+      '/route',
+      'POST'
+    )
+  })
+
+  test('route returns 400 for parse errors', async () => {
+    expect.assertions(1)
+    const router = Router()
+    router.post('/route', jest.fn())
+    const result = await router.route(
+      { body: 'name', headers: { 'Content-Type': 'application/json' } } as any,
+      {},
+      '/route',
+      'POST'
+    )
+    expect(result.response.statusCode).toBe(400)
+  })
+
+  test('route does not parse body with option', async () => {
+    expect.assertions(1)
+    const router = Router({ parseBody: false })
+    router.post('/route', (async ({ body }) => {
+      expect(body).toBe(JSON.stringify({ name: 'tim' }))
+    }) as any)
+    await router.route(
+      {
+        body: JSON.stringify({ name: 'tim' }),
+        headers: { 'Content-Type': 'application/json' },
+      } as any,
+      {},
+      '/route',
+      'POST'
+    )
+  })
+
+  test('route decodes parameters with default option', async () => {
+    expect.assertions(2)
+    const router = Router()
+    router.post('/route', (async ({ pathParameters, queryStringParameters }) => {
+      expect(pathParameters.name).toBe('tim kye')
+      expect(queryStringParameters.name).toBe('tim kye')
+    }) as any)
+    await router.route(
+      {
+        pathParameters: { name: 'tim%20kye' },
+        queryStringParameters: { name: 'tim%20kye' },
+      } as any,
+      {},
+      '/route',
+      'POST'
+    )
+  })
+
+  test('route does not decode parameters with option', async () => {
+    expect.assertions(2)
+    const router = Router({ decodeEvent: false })
+    router.post('/route', (async ({ pathParameters, queryStringParameters }) => {
+      expect(pathParameters.name).toBe('tim%20kye')
+      expect(queryStringParameters.name).toBe('tim%20kye')
+    }) as any)
+    await router.route(
+      {
+        pathParameters: { name: 'tim%20kye' },
+        queryStringParameters: { name: 'tim%20kye' },
+      } as any,
+      {},
+      '/route',
+      'POST'
+    )
+  })
+
+  test('traceId is created', async () => {
+    expect.assertions(1)
+    const router = Router()
+    router.post('/route', jest.fn())
+    const result = await router.route({} as any, {}, '/route', 'POST')
+    expect(result.response.headers['X-Correlation-Id']).toBeTruthy()
```

Comment:
is it worth to check/lock the format of the created traceid? (uuid)

Reply:
No, some apps use nanoids